File: commonlibrary/src/main/java/com/ayvytr/commonlibrary/FileUtil.java
package com.ayvytr.commonlibrary;

import android.os.Build;
import android.os.Environment;
import android.os.StatFs;
import android.support.annotation.RequiresApi;

/**
 * @author admin
 */
public class FileUtil {

    @RequiresApi(api = Build.VERSION_CODES.JELLY_BEAN_MR2)
    public static long getSDFreeSpace() {
        StatFs stat = new StatFs(Environment.getExternalStorageDirectory().getAbsolutePath());
        long blockSize = stat.getBlockSizeLong();
        long availableBlocks = stat.getAvailableBlocksLong();
        return availableBlocks * blockSize;
    }
}

File: commonlibrary/src/main/java/com/ayvytr/commonlibrary/FileUtil.kt
package com.ayvytr.commonlibrary

import android.os.Build
import android.os.Environment
import android.os.StatFs
import android.support.annotation.RequiresApi

/**
 * @author admin
 */
object FileUtil {

    val sdFreeSpace: Long
        @RequiresApi(api = Build.VERSION_CODES.JELLY_BEAN_MR2)
        get() {
            val stat = StatFs(Environment.getExternalStorageDirectory().absolutePath)
            val blockSize = stat.blockSizeLong
            val availableBlocks = stat.availableBlocksLong
            return availableBlocks * blockSize
        }
}
